### DISCUSSION TITLE
New Tool - mini

### DISCUSSION THREAD

**Discussion by WendellAdriel** [minicli/minicli#75] [Ideas]
@erikaheidi @JustSteveKing 

As you know, Laravel has a tool called Artisan that has a lot of amazing commands.

I was thinking on building a command using minicli itself that would be called `mini`, that would be something like Artisan.

I’m thinking on building only one command called create that would be used to create commands, themes and services like:

```bash
mini create command TestController
mini create service TestService
mini create theme TestTheme
```

After this first release we can create more commands to list commands, list services, list configurations, etc.

If you like this idea I’d love to work on this after we “release” miniterm!

**Comment by JustSteveKing** [comment]
I love this idea! I think we could potentially do a lot with this

**Reply by WendellAdriel** [reply]
Yeah, for the future this can be a really powerful tool since it will enable a lot of possibilities!

**Reply by JustSteveKing** [reply]
```bash
mini system
```

To show information about your Minicli app 🔥

**Reply by WendellAdriel** [reply]
Let’s make it happen!
I’ll be starting to work on `mini` ASAP and I’ll send updates here!

**Comment by WendellAdriel** [comment]
@erikaheidi @JustSteveKing one QQ on this.
Should this new tool, mini be put in the `minicli/minicli` repository or should it be a new repository like `minicli/mini` that will be a default dependency for `minicli/application` like the `minicli/command-help`?

If this should be a new repository, can any of you create it so I can start working directly on a repository in the minicli org or should I start on a personal repository like miniterm and then we move to the minicli org?

**Reply by JustSteveKing** [reply]
I think start on a private repo for a proof of concept, then we can transfer it to the org later?

minicli/mini would be cool!

**Reply by WendellAdriel** [reply]
Nice, I'll do that, thanks!

**Reply by erikaheidi** [reply]
Yess I was just going to propose that. As an external command, this will be super helpful! then you can still create a minimal app if you want, but the application templates would add this command by default just as it is with the help command currently.

**Reply by WendellAdriel** [reply]
Amazing! Let’s go for it! I’ll start working on this!
